web-console/src/lib/functions/felderaRelation.ts

```diff
@@ -0,0 +1,28 @@
+import { Field, Relation } from '$lib/services/manager'
+import invariant from 'tiny-invariant'
+
+export type CaseDependentName = {
+  case_sensitive?: boolean | undefined
+  name: string
+}
+
+export const quotifyRelationName = (relation: Relation) =>
+  relation.case_sensitive ? `"${relation.name}"` : relation.name
+
+export const caseDependentName = (quotedRelationName: string) => {
+  const name = /\"?(\w+)\"?/.exec(quotedRelationName)?.[1]
+  invariant(name, `Invalid relation name parsed: ${name}`)
+  return {
+    name,
+    case_sensitive: /\"/.test(quotedRelationName)
```

Comment:
You already checked before, if you write simpler code you don't need to check a second time.
Plus, this check is not semantically equivalent with the previous one.

Reply:
I updated the code to lax 
```
    name: quotedRelationName.replaceAll('"', ''),
    case_sensitive: /\"/.test(quotedRelationName)
```
Assuming the argument is well formed the semantics are equivalent; If the need for a stricter check arises I'll refactor this.

Reply:
These are still not equivalent.
You should not remove all quotes, only the first and the last. Quotes are legal inside a name too. Hope you have tests with that. But I didn't see any

Reply:
I was not aware quotes are a valid part of a name

Reply:
They are in some sql dialects
https://learn.microsoft.com/en-us/sql/odbc/microsoft/table-name-limitations?view=sql-server-ver16

Reply:
Please see Lalith's comment: https://github.com/feldera/feldera/issues/1301#issuecomment-1920195980 ; He suggests allowed symbols should/will be alphanumeric, hyphens and, I suppose, space

Reply:
The whole point of using sql is that people can bring existing programs. We should impose as few constraints as possible

Reply:
Currently neither of these compile:
```
CREATE TABLE `prod"uct`
CREATE TABLE 'prod"uct'
CREATE TABLE prod"uct
```
Would you be comfortable leaving this simple code as-is which is valid for two currently supported cases:

- Double quotes are not used in a name

- The name is wrapped in double quotes

, and creating a separate issue for support of using double quotes inside the name when it is wrapped in back quotes (`)?

Reply:
sure